src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesSubTreeProviderBase.cs

```diff
@@ -0,0 +1,618 @@
+                                ProjectSubscriptionService.SharedFoldersSource.SourceBlock.SyncLinkOptions(),
+                                actionBlock);
+                        }
+                    },
+                    registerFaultHandler: true);
+            }
+        }
+
+
+        /// <summary>
+        /// Disposes this instance.
+        /// </summary>
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _projectSubscriptionLink?.Dispose();
+                _projectSyncLink?.Dispose();
+            }
+        }
+
+        /// <summary>
+        /// Handles changes to the references items in the project and updates the project tree.
+        /// </summary>
+        /// <param name="e">A description of the changes made to the project.</param>
+        private void ProjectSubscriptionService_Changed(IProjectVersionedValue<
+                                                            Tuple<IProjectSubscriptionUpdate,
+                                                                  IProjectCatalogSnapshot,
+                                                                  IProjectSharedFoldersSnapshot>> e)
+        {
+
+            var dependenciesChange = ProcessDependenciesChanges(e.Value.Item1, e.Value.Item2);
+
+            // process separatelly shared projects changes
+            ProcessSharedProjectImportNodes(e.Value.Item3, dependenciesChange);
+
+            // Apply dependencies changes to actual RootNode children collection
+            // remove first nodes from actual RootNode
+            dependenciesChange.RemovedNodes.ForEach(RootNode.RemoveChild);
+
+            ProcessDuplicatedNodes(dependenciesChange);
+
+            dependenciesChange.AddedNodes.ForEach(RootNode.AddChild);
+
+            OnDependenciesChanged(dependenciesChange.GetDiff(), e);
+        }
+
+        public virtual IDependencyNode GetDependencyNode(DependencyNodeId nodeId)
+        {
+            Requires.NotNull(nodeId, nameof(nodeId));
+
+            return RootNode.Children.FirstOrDefault(x => x.Id.Equals(nodeId));
+        }
+
+        public IDependencyNode GetDependencyNode(string nodeIdString)
+        {
+            Requires.NotNullOrEmpty(nodeIdString, nameof(nodeIdString));
+
+            var id = DependencyNodeId.FromString(nodeIdString);
+            return GetDependencyNode(id);
+        }
+
+        public virtual Task<IEnumerable<IDependencyNode>> SearchAsync(IDependencyNode node, string searchTerm)
+        {
+            return Task.FromResult<IEnumerable<IDependencyNode>>(null);
+        }
+
+        protected virtual IDependencyNode CreateRootNode()
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Creates a specific node type for given provider
+        /// </summary>
+        protected virtual IDependencyNode CreateDependencyNode(string itemSpec,
```

Comment:
>CreateDependencyNode [](start = 42, length = 20)

This sounds like this should be abstract - i.e. the class should be abstract with these virtual methods that throw notimpl or are empty should be abstract to force an implementation?